cmd/util/cmd/read-badger/cmd/transactions.go

```diff
@@ -39,3 +53,162 @@ var transactionsCmd = &cobra.Command{
+				if err != nil {
+					panic(fmt.Sprintf("error while constructing system chunk tx: %s", err))
+				}
+				systemTxID = systemTx.ID()
+
+				hasSystemTx = true
+			}
+
+			blockID := header.ID()
+
+			block, err := storages.Blocks.ByID(blockID)
+			if err != nil {
+				panic(fmt.Sprintf("header %s if here but not block", blockID.String()))
+			}
+
+			missingHeights[header.Height] = struct{}{}
+			allHeights[header.Height] = struct{}{}
+
+			txIndex := uint32(0)
+
+			txs := make([]flow.Identifier, 0)
+
+			for _, guarantee := range block.Payload.Guarantees {
+				lightCollection, err := storages.Collections.LightByID(guarantee.CollectionID)
+				if err != nil {
+
+					log.Warn().Msgf("cannot get light collection %s for block %s (%d), trying full collection: %s", guarantee.CollectionID, blockID.String(), header.Height, err)
+
+					collection, err := storages.Collections.ByID(guarantee.CollectionID)
+
+					if err != nil {
+						panic(fmt.Sprintf("cannot get light and normal collection %s for block %s (%d): %s", guarantee.CollectionID, blockID.String(), header.Height, err))
+					}
+					light := collection.Light()
+					lightCollection = &light
+				}
+
+				txs = append(txs, lightCollection.Transactions...)
+			}
+
+			// add system tx
+			txs = append(txs, systemTxID)
+
+			for _, txID := range txs {
+
+				if _, has := megaMap[txID]; !has {
+					megaMap[txID] = make(map[flow.Identifier]struct{}, 0)
+				}
+
+				if _, has := megaMap[txID][blockID]; has {
+					panic(fmt.Sprintf("duplicated tx %s in block %s", txID.String(), blockID.String()))
+				}
+
+				megaMap[txID][blockID] = struct{}{}
+
+				_, err := storages.TransactionResults.ByBlockIDTransactionIndex(blockID, txIndex)
+				if err != nil {
+					if errors.Is(err, storage.ErrNotFound) {
+						transactionResult, err := storages.TransactionResults.ByBlockIDTransactionID(blockID, txID)
+
+						if flagFixTransactionResultsIndex {
+							if err == nil {
+								result := operation.BatchIndexTransactionResult(blockID, txIndex, transactionResult)
+								err = result(batch.GetWriter())
+								if err != nil {
+									panic(fmt.Sprintf("cannot batch index tx results by  (block_id, tx_index) (%s, %d): %s", blockID.String(), txIndex, err))
+								}
+							}
+						}
+						delete(missingHeights, header.Height) // assume existing entries means whole block is mapped for a height
+
+						indexNewEntries++
+					} else {
+						panic(fmt.Sprintf("error while querying transaction result by (block_id, tx_index) (%s, %d): %s", blockID.String(), txIndex, err))
+					}
+				} else {
+					delete(missingHeights, header.Height) // assume existing entries means whole block is mapped for a height
+					indexExistingEntries++
+				}
+
+				totalTxs++
+				txIndex++
+			}
+
+			blockNo++
+
+			if blockNo%10000 == 0 {
+				log.Info().Int("blocks", blockNo).Int("total_tx", totalTxs).Msg("processing progress")
+			}
```

Comment:
What dies this check do? it is just logging for every 10k blocks?

Reply:
Yes, it takes quite long to process (20-30mins)